### ISSUE TITLE
`@stylistic/indent`: single line comment behavior

### ISSUE THREAD

**Issue by cbratschi** [eslint-stylistic/eslint-stylistic#1018]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The following code passes but indentation should be 4 spaces according to my configuration:

```JavaScript
function a() {
    //this ...
//text
}
```

If "//text" is indented by a single space it gets reported as an error (4 spaces expected):

```JavaScript
function a() {
    //this ...
 //text
}
```

This case is also reported correctly (expected 4 spaces):

```JavaScript
function a() {
    //this ...

//text

}
```

That one not:

```JavaScript
function a() {
    //this ...
//text
    const a = 5;

}
```

This rule does not handle all comment cases.

### Reproduction

See examples above.

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
